src/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Collect source files for library
+set(rcm-sources
+        rcm.f90
+        rcm_kinds.f90
+)
+
+
+# Build library
+add_library(rcm-f90 ${rcm-sources})
+
+# Store generated mod-files in a separate folder
+set(moddir "${CMAKE_CURRENT_BINARY_DIR}/mod")
```

Comment:
Is this not set by the parent CMakeLists.txt? Also, as there please use `include/` to avoid changing other projects.